test_spack.py

```diff
@@ -5,44 +5,58 @@
 import sys
 import subprocess
 import unittest
+import asyncio
 
 all_machines = {'daint', 'tsa'}
 
-
-def run(command: str, cwd='.'):
-    setup = ''
-    if command.startswith('spack'):
-        setup = f'source spack/share/spack/setup-env.sh &&'
-
-    subprocess.run(f'{setup} cd {cwd} && {command}', check=True, shell=True)
-
-
 # For each spack test there should be at least one line of comment stating
 # why this spack command is tested and/or where this spack command is used.
 # Otherwise this may apply:
 # “All Your Tests are Terrible..." - Titus Winters & Hyrum Wright
 # https://www.youtube.com/watch?v=u5senBJUkPc&ab_channel=CppCon
 
 
-class AtlasTest(unittest.TestCase):
+class TestCase(unittest.TestCase):
+
+    def Run(self, command: str, cwd='.', parallel=False):
+        setup = ''
+        if command.startswith('spack'):
+            setup = f'source spack/share/spack/setup-env.sh ; '
+
+        srun = ''
+        if parallel:
+            if machine == 'tsa':
+                srun = 'srun -c 16 -t 01:00:00'
```

Comment:
Can we add a srun for Daint here?

Something like
`srun -A g110 -C gpu -t 01:00:00`